app/assets/javascripts/discourse/app/controllers/reorder-categories.js

```diff
@@ -63,56 +63,102 @@ export default Controller.extend(ModalFunctionality, Evented, {
     this.notifyPropertyChange("categoriesBuffered");
   },
 
-  move(category, direction) {
-    let otherCategory;
+  countDescendants(category) {
+    // Recursive function to count subcategories
+    if (!category.get("isParent")) {
```

Comment:
Thank you. 

>It would be nice to see if category.subcategories is always set (never undefined or null). In which case, you could get rid of the base case completely.

I tried removing the check for `category.subcategories` being set and got the following error:
```
Uncaught TypeError: category.get(...) is undefined
```
It seems like it is sometimes undefined.